src/main/java/org/stellar/sdk/Sep10Challenge.java

```diff
@@ -130,10 +132,25 @@ public static ChallengeTransaction readChallengeTransaction(String challengeXdr,
       throw new InvalidSep10ChallengeException("Operation should have a source account.");
     }
 
+    String matchedDomainName = null;
+    for (String homeDomain : domainNames) {
+      if ((homeDomain + " " + MANAGER_DATA_NAME_FLAG).equals(manageDataOperation.getName())) {
+        matchedDomainName = homeDomain;
+        break;
+      }
+    }
+
+    if (matchedDomainName == null) {
+      throw new InvalidSep10ChallengeException("The transaction's operation key name does not include the expected home domain.");
```

Comment:
Not too important but I would one of the following:
```suggestion
      throw new InvalidSep10ChallengeException("The transaction's operation key name does not include any expected home domain.");
```
```suggestion
      throw new InvalidSep10ChallengeException("The transaction's operation key name does not include one of the expected home domains.");
```